### PR TITLE
Fixed restart command

### PR DESCRIPTION
**Pull request by AutumnClove**
Please make sure these boxes are checked (put an 'x' inside them) **before** creating the pull request.

- [x] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.5/3.6

----

### Description
I already did a pull request for this but I messed up the indent sizes and couldn't figure out how to fix it. 
So I just did a new one .-.

### Related issues (if applicable)
#1493

### CODE REVIEW

File: musicbot/bot.py [hunk 1]
```diff
@@ -2546,9 +2546,22 @@ def check(message):
...
+        """
+        Usage: 
+            {command_prefix}restart
+
+        Restarts the bot, making it leave all voice channels, this also opens a new command prompt, with any saved changes to files.
+        """
         await self.safe_send_message(channel, "\N{WAVING HAND SIGN}")
         await self.disconnect_all_voice_clients()
-        raise exceptions.RestartSignal()
+        from sys import platform as _platform
```
Comment (jayktaylor) [inline musicbot/bot.py:2557]: bot.py already imports the sys module, you don't need to do this.

File: musicbot/bot.py [hunk 2]
```diff
@@ -2546,9 +2546,22 @@ def check(message):
...
+        if _platform.startswith("linux"):
+	    p = subprocess.Popen(runbot.linux.sh, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        elif _platform == "darwin":
+            p = subprocess.Popen('runbot_osx.command', creationflags=subprocess.CREATE_NEW_CONSOLE)
+        else:
+            p = subprocess.Popen('runbot.bat', creationflags=subprocess.CREATE_NEW_CONSOLE)
```
Comment (jayktaylor) [inline musicbot/bot.py:2563]: This isn't how subprocesses work. You're just going to end up creating multiple processes within the same process, and then you'll eventually run into memory errors.

File: musicbot/bot.py [hunk 3]
```diff
@@ -2546,9 +2546,22 @@ def check(message):
...
+        raise exceptions.TerminateSignal()
```
Comment (jayktaylor) [inline musicbot/bot.py:2564]: I don't think this point will ever actually be reached until the subprocesses have finished, which again is a problem that contributes to the memory issues mentioned beforehand.

**Review by jayktaylor — CHANGES_REQUESTED** [review]

**Review by MattBSG — CHANGES_REQUESTED** [review]
Subprocesses and threads are not a good solution because it can create huge problems especially pertaining to memory. It’s also really bad practice to make tens of sub-threads in cases a user may never actually close the bot. A completely different solution will be needed (think how ours is now, but add on to it)

**Pull request closed without merging**